code/modules/admin/verbs/debug.dm

```diff
@@ -655,17 +655,17 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
 
 	var/list/names = list()
 	names += "---- Space Ruins ----"
-	for(var/name in SSmapping.space_ruins_templates)
-		names[name] = list(SSmapping.space_ruins_templates[name], ZTRAIT_SPACE_RUINS, list(/area/space))
+	for(var/name in SSmapping.themed_ruins[RUIN_TYPE_SPACE])
+		names[name] = list(SSmapping.themed_ruins[RUIN_TYPE_SPACE][name], ZTRAIT_SPACE_RUINS, list(/area/space))
 	names += "---- Lava Ruins ----"
-	for(var/name in SSmapping.lava_ruins_templates)
-		names[name] = list(SSmapping.lava_ruins_templates[name], ZTRAIT_LAVA_RUINS, list(/area/lavaland/surface/outdoors/unexplored))
+	for(var/name in SSmapping.themed_ruins[RUIN_TYPE_SPACE])
+		names[name] = list(SSmapping.themed_ruins[RUIN_TYPE_LAVA][name], ZTRAIT_LAVA_RUINS, list(/area/lavaland/surface/outdoors/unexplored))
 	names += "---- Ice Ruins ----"
-	for(var/name in SSmapping.ice_ruins_templates)
-		names[name] = list(SSmapping.ice_ruins_templates[name], ZTRAIT_ICE_RUINS, list(/area/icemoon/surface/outdoors/unexplored, /area/icemoon/underground/unexplored))
+	for(var/name in SSmapping.themed_ruins[RUIN_TYPE_ICE])
+		names[name] = list(SSmapping.themed_ruins[RUIN_TYPE_ICE][name], ZTRAIT_ICE_RUINS, list(/area/icemoon/surface/outdoors/unexplored, /area/icemoon/underground/unexplored))
 	names += "---- Ice Underground Ruins ----"
-	for(var/name in SSmapping.ice_ruins_underground_templates)
-		names[name] = list(SSmapping.ice_ruins_underground_templates[name], ZTRAIT_ICE_RUINS_UNDERGROUND, list(/area/icemoon/underground/unexplored))
+	for(var/name in SSmapping.themed_ruins[RUIN_TYPE_ICE_UNDERGROUND])
+		names[name] = list(SSmapping.themed_ruins[RUIN_TYPE_ICE_UNDERGROUND][name], ZTRAIT_ICE_RUINS_UNDERGROUND, list(/area/icemoon/underground/unexplored))
```

Comment:
How close are the z traits to being the same kind of info as the theme type

Reply:
nearly identical wtf im just gonna use that